nifi-extension-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/s3/CopyS3Object.java

```diff
@@ -0,0 +1,147 @@
+            .displayName("Destination Key")
+            .description("The target key in the target bucket")
+            .defaultValue("")
+            .build();
+
+    public static final List<PropertyDescriptor> properties = List.of(
+            SOURCE_BUCKET,
+            SOURCE_KEY,
+            DESTINATION_BUCKET,
+            DESTINATION_KEY,
+            AWS_CREDENTIALS_PROVIDER_SERVICE,
+            S3_REGION,
+            TIMEOUT,
+            FULL_CONTROL_USER_LIST,
+            READ_USER_LIST,
+            WRITE_USER_LIST,
+            READ_ACL_LIST,
+            WRITE_ACL_LIST,
+            CANNED_ACL,
+            OWNER,
+            SSL_CONTEXT_SERVICE,
+            ENDPOINT_OVERRIDE,
+            SIGNER_OVERRIDE,
+            S3_CUSTOM_SIGNER_CLASS_NAME,
+            S3_CUSTOM_SIGNER_MODULE_LOCATION,
+            PROXY_CONFIGURATION_SERVICE);
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return properties;
+    }
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        final AmazonS3Client s3;
+        try {
+            s3 = getS3Client(context, flowFile.getAttributes());
+        } catch (Exception e) {
+            getLogger().error("Failed to initialize S3 client", e);
+            flowFile = session.penalize(flowFile);
+            session.transfer(flowFile, REL_FAILURE);
+            return;
+        }
+
+        final String sourceBucket = context.getProperty(SOURCE_BUCKET).evaluateAttributeExpressions(flowFile).getValue();
+        final String sourceKey = context.getProperty(SOURCE_KEY).evaluateAttributeExpressions(flowFile).getValue();
+        final String targetBucket = context.getProperty(DESTINATION_BUCKET).evaluateAttributeExpressions(flowFile).getValue();
+        final String targetKey = context.getProperty(DESTINATION_KEY).evaluateAttributeExpressions(flowFile).getValue();
+
+        try {
+            CopyObjectRequest request = new CopyObjectRequest(sourceBucket, sourceKey, targetBucket, targetKey);
+            AccessControlList acl = createACL(context, flowFile);
+            if (acl != null) {
+                request.setAccessControlList(acl);
+            }
+            CannedAccessControlList cannedAccessControlList = createCannedACL(context, flowFile);
+
+            if (cannedAccessControlList != null) {
+                request.setCannedAccessControlList(cannedAccessControlList);
+            }
+
+            s3.copyObject(request);
+            session.getProvenanceReporter().send(flowFile, getTransitUrl(targetBucket, targetKey));
+
+            session.transfer(flowFile, REL_SUCCESS);
+        } catch (final AmazonClientException e) {
+            flowFile = extractExceptionDetails(e, session, flowFile);
+            getLogger().error("Failed to copy S3 object", e);
```

Comment:
Recommend including the source bucket and key:
```suggestion
            getLogger().error("Failed to copy S3 object from Bucket [{}] Key [{}]", sourceBucket, sourceKey, e);
```